### PR TITLE
[docs][PLAT-11684] Instructions for creating common TServer service

### PR DESCRIPTION
**Pull request by bhavin192**
Preview: https://deploy-preview-20283--infallible-bardeen-164bc9.netlify.app/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes/#create-common-tserver-service-for-zones

### CODE REVIEW

**Comment by bhavin192** [comment]
@ddhodge Sanketh has seen these changes, so you can review it whenever you can.

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md [hunk 1]
```diff
@@ -159,3 +159,47 @@ You can create a connection to a node as follows:
...
 
 For information on how to connect to the universe from the Kubernetes cluster, as well as remotely, see [Connect YugabyteDB clusters](../../../deploy/kubernetes/clients/#connect-tls-secured-yugabytedb-cluster-deployed-by-helm-charts).
+
+### Create common TServer service for zones
+
+By default each zone gets its own TServer service which can be used to connect to the universe. Optionally, you can create a common service across these zones as follows. 
+
+Note that this requires all the zone deployments to be in the same namespace.
+
+1. Set the following Kubernetes overrides to add universe name label on the TServer pods. You can do this during universe creation or by modifying the Kubernetes overrides of an existing universe.
```
Comment (iSignal) [inline docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md:169]: Can we link to the overrides doc section here?

**Review by iSignal — APPROVED** [review]

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md [hunk 2]
```diff
@@ -159,3 +159,47 @@ You can create a connection to a node as follows:
...
+
+Note that this requires all the zone deployments to be in the same namespace.
+
+1. Set the following Kubernetes overrides to add universe name label on the TServer pods. You can do this [during universe creation](#configure-helm-overrides) or by [modifying the Kubernetes overrides](../../manage-deployments/edit-helm-overrides/) of an existing universe.
+   ```yaml
+   tserver:
+     podLabels:
+       universe-name: yb-k8s
+   ```
+1. Save following block as `yb-tserver-common-service.yaml` file. You can modify the service type, annotations, and the label selector according to your needs.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md:175]: Where does the file go? Is it somehow assumed?

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md [hunk 3]
```diff
@@ -159,3 +159,47 @@ You can create a connection to a node as follows:
 ## Connect to the universe
 
 For information on how to connect to the universe from the Kubernetes cluster, as well as remotely, see [Connect YugabyteDB clusters](../../../deploy/kubernetes/clients/#connect-tls-secured-yugabytedb-cluster-deployed-by-helm-charts).
+
+### Create common TServer service for zones
+
+By default each zone gets its own TServer service which can be used to connect to the universe. Optionally, you can create a common service across these zones as follows.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md:165]:
```suggestion
By default, each zone has its own YB-TServer service, and you can use this service to connect to the universe. Optionally, you can create a common service across all zones as follows. 
```

Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md:165]: If you create the common service, do the individual ones remain? What is the benefit?

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md [hunk 4]
```diff
@@ -159,3 +159,47 @@ You can create a connection to a node as follows:
...
 
 For information on how to connect to the universe from the Kubernetes cluster, as well as remotely, see [Connect YugabyteDB clusters](../../../deploy/kubernetes/clients/#connect-tls-secured-yugabytedb-cluster-deployed-by-helm-charts).
+
+### Create common TServer service for zones
+
+By default each zone gets its own TServer service which can be used to connect to the universe. Optionally, you can create a common service across these zones as follows. 
+
+Note that this requires all the zone deployments to be in the same namespace.
+
+1. Set the following Kubernetes overrides to add universe name label on the TServer pods. You can do this [during universe creation](#configure-helm-overrides) or by [modifying the Kubernetes overrides](../../manage-deployments/edit-helm-overrides/) of an existing universe.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md:169]:
```suggestion
1. Set the following Kubernetes overrides to add the universe-name label on the TServer pods. You can do this when you [create the universe](#configure-helm-overrides) or by [modifying the Kubernetes overrides](../../manage-deployments/edit-helm-overrides/) of an existing universe.

```

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md (hunk 2 above)
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md:175]:
```suggestion

1. Save the following to a file named `yb-tserver-common-service.yaml`. You can customize the service type, annotations, and the label selector as required.

```

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md [hunk 5]
```diff
@@ -159,3 +159,47 @@ You can create a connection to a node as follows:
...
+     ports:
+     # Modify the ports if using non-standard ports.
+     - name: tcp-yedis-port
+       port: 6379
+     - name: tcp-yql-port
+       port: 9042
+     - name: tcp-ysql-port
+       port: 5433
+   ```
+1. Run the following command to create the service in the universe's namespace which is `yb-prod-yb-k8s` in our case.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md:201]:
```suggestion

1. Run the following command to create the service in the universe's namespace (`yb-prod-yb-k8s` in this example).

```

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md [hunk 6]
```diff
@@ -159,3 +159,47 @@ You can create a connection to a node as follows:
...
+   ```sh
+   $ kubectl apply -f yb-tserver-common-service.yaml -n yb-prod-yb-k8s
+   ```
+1. You can access the YugabyteDB at `yb-k8s-common-tserver.yb-prod-yb-k8s.svc.cluster.local`.
```
Comment (ddhodge) [inline docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md:205]:
```suggestion

After the overrides are applied, in this example you would access the universe at `yb-k8s-common-tserver.yb-prod-yb-k8s.svc.cluster.local`.

```

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md (hunk 2 above)
Comment (bhavin192) [reply docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md]: We use this temporary file to save the block, make any modifications, and then apply it in the next step. 
After that the we don't care what happens with the file basically.

File: docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md (hunk 3 above)
Comment (bhavin192) [reply docs/content/preview/yugabyte-platform/create-deployments/create-universe-multi-zone-kubernetes.md]: Mentioned that it is an additional service. 

About the benefits, I'm not sure how to write those here:
When you have a single service, your clients can connect to it and don't have to worry about individual TServers from different zones. If some zone's TServer(s) are down the request will be served by other zones TServer as the service takes care of load balancing. 

For now I have modified the sentence to say we create an highly available service with these steps.

**Review by ddhodge — APPROVED** [review]

**Pull request merged**